rust/scx_utils/src/netdev.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+
+// This software may be used and distributed according to the terms of the
+// GNU General Public License version 2.
+
+use std::collections::BTreeMap;
+use std::fs;
+use std::path::Path;
+
+use crate::misc::read_file_usize;
+use crate::Cpumask;
+use anyhow::Result;
+
+#[derive(Debug, Clone, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub struct NetDev {
+    pub iface: String,
+    pub node: usize,
+    pub irqs: BTreeMap<usize, Cpumask>,
+    pub irq_hints: BTreeMap<usize, Cpumask>,
```

Comment:
Maybe worthwhile to tighten up data model. For example, `irq_hints` is exported from driver to userspace, so this should probably be a read-only getter that returns a const reference to private member (and not available as a publicly modifiable member). Same goes for `iface` and `node`.

`irqs` looks like it's doing double duty here - `read_netdevs()` uses it to return initial config. Then callers will modify the member and trigger a write thru `apply_cpumasks`. Perhaps better to make `irqs` a private member and then have a getter/setter pair, where getter returns whatever is cached and setter is a write thru cache. Setter could just be `apply_cpumasks()` accepting a cpumask.